lib/segment/src/types.rs

```diff
@@ -628,28 +634,48 @@ impl SegmentConfig {
     }
 
     pub fn distance(&self, vector_name: &str) -> Option<Distance> {
-        self.vector_data
+        let distance = self
+            .vector_data
             .get(vector_name)
-            .map(|config| config.distance)
+            .map(|config| config.distance);
+        if distance.is_none() {
+            self.sparse_vector_data
+                .get(vector_name)
+                .map(|_config| Distance::Dot)
```

Comment:
Don't we have a constant for `SparseDistance` yet?

Reply:
Added constant and applied for sparse vector storage too